### PR TITLE
refactored to remove exported gnatsd dependency.

### LINKED ISSUE #314: Remove gnatsd dependency
`gnatsd` is included as a dependency because of `test.go` exported functions.

Remove the dependency to eliminate any circular dependencies on the go-nats library.

### PR DESCRIPTION
**Pull request by petemiron**
Resolves #314 

- Moved common functions to unexported `helper_test.go`
- Added `test/main.go` with simple `TestMain`
- Required moving `encoders` tests to `test` package

dependencies prior to change:
```
go list -f '{{ join .Deps "\n" }}' ./... | sort | uniq | grep github.com
github.com/golang/protobuf/proto
github.com/nats-io/gnatsd/conf
github.com/nats-io/gnatsd/logger
github.com/nats-io/gnatsd/server
github.com/nats-io/gnatsd/server/pse
github.com/nats-io/gnatsd/test
github.com/nats-io/gnatsd/util
github.com/nats-io/gnatsd/vendor/github.com/nats-io/nuid
github.com/nats-io/gnatsd/vendor/golang.org/x/crypto/bcrypt
github.com/nats-io/gnatsd/vendor/golang.org/x/crypto/blowfish
github.com/nats-io/go-nats
github.com/nats-io/go-nats/bench
github.com/nats-io/go-nats/encoders/builtin
github.com/nats-io/go-nats/util
github.com/nats-io/nuid
```

dependencies after:
```
go list -f '{{ join .Deps "\n" }}' ./... | sort | uniq | grep github.com
github.com/golang/protobuf/proto
github.com/nats-io/go-nats
github.com/nats-io/go-nats/bench
github.com/nats-io/go-nats/encoders/builtin
github.com/nats-io/go-nats/util
github.com/nats-io/nuid
```

### CODE REVIEW

File: scripts/cov.sh [hunk 1]
```diff
@@ -7,6 +7,8 @@ go test -v -covermode=atomic -coverprofile=./cov/nats.out
 go test -v -covermode=atomic -coverprofile=./cov/builtin.out ./encoders/builtin
 go test -v -covermode=atomic -coverprofile=./cov/protobuf.out ./encoders/protobuf
 go test -v -covermode=atomic -coverprofile=./cov/test.out -coverpkg=github.com/nats-io/go-nats ./test
+go test -v -covermode=atomic -coverprofile=./cov/test.out -coverpkg=github.com/nats-io/go-nats/encoders/builtin ./test -run EncBuiltin
```
Comment (kozlovic) [inline scripts/cov.sh]: Don't reuse `test.out` otherwise you lose output from previous run. User `test1.out`.

File: scripts/cov.sh [hunk 2]
```diff
@@ -7,6 +7,8 @@ go test -v -covermode=atomic -coverprofile=./cov/nats.out
 go test -v -covermode=atomic -coverprofile=./cov/builtin.out ./encoders/builtin
```
Comment (kozlovic) [inline scripts/cov.sh]: Remove this line and next, since there is no test in those packages.

File: scripts/cov.sh [hunk 3]
```diff
@@ -7,6 +7,8 @@ go test -v -covermode=atomic -coverprofile=./cov/nats.out
...
 go test -v -covermode=atomic -coverprofile=./cov/protobuf.out ./encoders/protobuf
 go test -v -covermode=atomic -coverprofile=./cov/test.out -coverpkg=github.com/nats-io/go-nats ./test
+go test -v -covermode=atomic -coverprofile=./cov/test.out -coverpkg=github.com/nats-io/go-nats/encoders/builtin ./test -run EncBuiltin
+go test -v -covermode=atomic -coverprofile=./cov/test.out -coverpkg=github.com/nats-io/go-nats/encoders/protobuf ./test -run EncProto
```
Comment (kozlovic) [inline scripts/cov.sh]: Same here, call it `test2.out`.

File: test/protobuf_test.go [hunk 4]
```diff
@@ -1,29 +1,28 @@
...
 
 	"github.com/nats-io/go-nats"
-	"github.com/nats-io/go-nats/test"
 
 	"github.com/nats-io/go-nats/encoders/protobuf"
 	pb "github.com/nats-io/go-nats/encoders/protobuf/testdata"
 )
 
-const TEST_PORT = 8068
+// const TEST_PORT = 8068
```
Comment (wallyqs) [inline test/protobuf_test.go]: comment can be removed since already defined somewhere else

**Comment by kozlovic** [comment]
LGTM

**Pull request merged**